federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -211,3 +212,20 @@ func NewCluster(name string, readyStatus api_v1.ConditionStatus) *federation_api
 		},
 	}
 }
+
+// Ensure a key is in the store before returning (or timeout w/ error)
+func WaitForStoreUpdate(store util.FederatedReadOnlyStore, clusterName, key string, timeout time.Duration) error {
+	timeoutEnd := time.Now().Add(timeout)
```

Comment:
The right idea, but please use wait.Poll from https://github.com/kubernetes/kubernetes/blob/master/pkg/util/wait/wait.go#L158